torch/ao/sparsity/experimental/activation_sparsifier/activation_sparsifier.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict, Any
+from collections import defaultdict
+from torch import nn
+
+__all__ = ['ActivationSparsifier']
+
+
+class ActivationSparsifier:
+    r"""
+    The Activation sparsifier class aims to sparsify/prune activations in a neural
+    network. The idea is to attach the sparsifier to a layer (or layers) and it
+    zeroes out the activations based on the mask_fn (or sparsification function)
+    input by the user.
+    The mask_fn is applied once all the inputs are aggregated and reduced i.e.
+    mask = mask_fn(reduce_fn(aggregate_fn(activations)))
+
+    Note::
+        The sparsification mask is computed on the input **before it goes through the attached layer**.
+
+    Args:
+        model (nn.Module)
+            The model whose layers will be sparsified. The layers that needs to be
+            sparsified should be added separately using the register_layer() function
+
+        aggregate_fn (Optional, function type)
+            default aggregate_fn that is used if not specified while registering the layer.
+            specifies how inputs should be aggreagated over time.
+
+        reduce_fn (Optional, function type)
```

Comment:
its hard to understand what the aggregate and reduce_fn do, also if there are defaults, shouldn't those be specified?